db_whitebox_test.go

```diff
@@ -112,6 +119,194 @@ func TestMethodPage(t *testing.T) {
 	}
 }
 
+func TestTx_Check_CorruptPage(t *testing.T) {
+	testCases := []struct {
+		name      string
+		bucketKey string
+		isPanic   bool
+		isErr     bool
+	}{
+		{
+			name:      "no corrupted page",
+			bucketKey: "testBucket",
+			isPanic:   false,
+			isErr:     false,
+		},
+		{
+			name:      "corrupt page by violating btree invariant",
+			bucketKey: "testBucket",
+			isPanic:   false,
+			isErr:     true,
+		},
+		{
+			name:      "corrupt page by dumping random bytes",
+			bucketKey: "testBucket",
+			isPanic:   true,
+			isErr:     false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			fileName, err := prepareData(t)
+			require.NoError(t, err)
+
+			t.Logf("Creating db file '%v'.", fileName)
+			db, err := Open(fileName, 0666, &Options{PageSize: 4096})
+			require.NoError(t, err)
+			defer func() {
+				require.NoError(t, db.Close())
+			}()
+
+			uErr := db.Update(func(tx *Tx) error {
+				t.Logf("Creating bucket '%v'.", tc.bucketKey)
+				b, bErr := tx.CreateBucketIfNotExists([]byte(tc.bucketKey))
+				require.NoError(t, bErr)
+				t.Logf("Generating random data in bucket '%v'.", tc.bucketKey)
+				generateSampleDataInBucket(t, b, 3)
+				return nil
+			})
+			require.NoError(t, uErr)
+
+			var victimPageId common.Pgid
+			if tc.isPanic || tc.isErr {
```

Comment:
@ahrtr @fuweid  i corrupt a page only if it is expected, otherwise i dont

might be ugly, but this is to cover the `no corruption` case 

looking for better suggestions 👍🏽